src/main/java/dk/kb/labsapi/ImageExport.java

```diff
@@ -531,4 +533,226 @@ private double convertInch1200ToPixels(double inch1200Value){
+        return docs;
+    }
+
+    /**
+     * Get metadata values for a given SolrDocument.
+     * The returned object contains metadata about a single page.
+     * @return an object containing metadata from a single page. metadata values are: pageUUID, pageWidth and pageHeight.
+     */
+    public FullPageMetadata streamMetadataForFullPage(SolrDocument doc) {
+        FullPageMetadata page = null;
+        try {
+            page = new FullPageMetadata(doc.get("pageUUID").toString(), (Long) doc.get("page_width"), (Long) doc.get("page_height"));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return page;
+    }
+
+    /**
+     * Get a stream of IllustrationMetadata from a given SolrDocument
+     * The returned stream contains metadata about each single illustration in the given SolrDocument.
+     * IllustrationMetadata objects contain the following values: String id, int x, int y, int w, int h, string pageUUID, int pageWidth, int pageHeight and imageURL.
+     * X and Y are coordinates, w = width and h = height. pageUUID, pageWidth and pageHeight are related to the page, which the illustration has been extracted from and imageURL is the URL where the illustration is available.
+     * @return a list of metadata objects consisting of the id, x, y, w, h, pageUUID, pageWidth and pageHeight values that are used to extract illustrations.
+     */
+    public Stream<IllustrationMetadata> streamMetadataForIllustrations(SolrDocument doc, HashSet<String> uniqueUUIDs) {
+        // TODO: This endpoint still returns some odd illustrations, which are clearly not illustrations nut flaws in the illustration boxes. However it works and these illustrations can be filtered away later by filtering small hights away
+        List<String> illustrationList = new ArrayList<>();
+        List<IllustrationMetadata> metadataList = new ArrayList<>();
+
+
+        // Extract metadata from SolrDocument
+        String pageUUID = doc.getFieldValue("pageUUID").toString();
+        String correctUUID = convertPageUUID(pageUUID);
+        if (!uniqueUUIDs.add(correctUUID)){
+            return null;
+        }
+        long pageWidth = (long) doc.getFieldValue("page_width");
+        long pageHeight = (long) doc.getFieldValue("page_height");
+        List<String> illustrations = (List<String>) doc.getFieldValue("illustration");
+        // Check if illustrations are present. If not, continue to next SolrDocument in list
+        if (illustrations == null) {
+            return Stream.empty();
+        }
+
+        return illustrations.stream().map(metadata -> new IllustrationMetadata(metadata, correctUUID, pageWidth, pageHeight));
+    }
+
+    /**
+     * Streams content of all metadatas given in input stream to an output stream that delivers a zip file of all images from the URLs.
+     * @param illustrationMetadata used to stream URLS from and construct filenames.
```

Comment:
Remember to update method signatures (the `illustrationMetadata` is no longer fitting) when the implementation changes.